File: backend/src/main/kotlin/com/cyborgstudios/gamebuilder/application/team/design/DesignersAgentService.kt
package com.cyborgstudios.gamebuilder.application.team.design

import com.cyborgstudios.gamebuilder.domain.model.Artifact
import com.cyborgstudios.gamebuilder.domain.model.ArtifactType
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class DesignersAgentService {

    fun createVisionArtifact(projectId: UUID, prompt: String): Artifact {
        val payload = """
            {"summary":"Vision generated for prompt: ${escapeJson(prompt)}"}
        """.trimIndent()

        return Artifact(
            id = UUID.randomUUID(),
            projectId = projectId,
            type = ArtifactType.VISION_DOC,
            payload = payload,
            createdAt = System.currentTimeMillis()
        )
    }

    private fun escapeJson(input: String): String =
        input.replace("\\", "\\\\").replace("\"", "\\\"")
}

File: backend/src/gamebuilder/team/design/application/designers_agent_service.py
from gamebuilder.team.design.application.port.design_agent_graph import DesignAgentGraph
from gamebuilder.team.design.domain.model import DesignTeamInput, DesignTeamOutput


class DesignersAgentService:
    def __init__(self, design_agent_graph: DesignAgentGraph) -> None:
        self._design_agent_graph = design_agent_graph

    def generate_initial_design(self, prompt: str) -> DesignTeamOutput:
        return self._design_agent_graph.run(DesignTeamInput(prompt=prompt))
